pkg/referrerstore/oras/cosign_test.go

```diff
@@ -0,0 +1,149 @@
+		input  common.Reference
+		output string
+		err    error
+	}{
+		{
+			input: common.Reference{
+				Path:   "localhost:5000/net-monitor",
+				Tag:    "v1",
+				Digest: getDigest("sha256:a0fc570a245b09ed752c42d600ee3bb5b4f77bbd70d8898780b7ab43454530eb"),
+			},
+			output: "localhost:5000/net-monitor:sha256-a0fc570a245b09ed752c42d600ee3bb5b4f77bbd70d8898780b7ab43454530eb.sig",
+			err:    nil,
+		},
+		{
+			input: common.Reference{
+				Path: "localhost:5000/net-monitor",
+				Tag:  "v1",
+			},
+			err: ErrNoCosignSubjectDigest,
+		},
+	}
+
+	for _, testcase := range testcases {
+		mutated, err := attachedImageTag(testcase.input, CosignSignatureTagSuffix)
+		if err != nil && !errors.Is(err, ErrNoCosignSubjectDigest) {
+			t.Fatalf("expected error to be %v, but got %v", testcase.err, err)
+		}
+		if mutated != testcase.output {
+			t.Fatalf("expected image tag to be %s, but got %s", testcase.output, mutated)
+		}
+	}
+}
+
+func TestGetCosignReferences(t *testing.T) {
+	ctx := context.Background()
+	testSubjectDigest := digest.FromString("test")
+	testCosignSubjectTag := fmt.Sprintf("%s-%s.sig", testSubjectDigest.Algorithm().String(), testSubjectDigest.Hex())
+	testCosignImageDigest := digest.FromString("test_cosign")
+	testcases := []struct {
+		name       string
+		subjectRef common.Reference
+		store      *orasStore
+		repository registry.Repository
+		output     *[]ocispecs.ReferenceDescriptor
+		err        error
+	}{
+		{
+			name: "failure: no subject digest",
+			subjectRef: common.Reference{
+				Path: "localhost:5000/net-monitor",
+				Tag:  "v1",
+			},
+			store:      &orasStore{},
+			repository: mocks.TestRepository{},
+			output:     nil,
+			err:        ErrNoCosignSubjectDigest,
+		},
+		{
+			name: "success: no cosign references",
+			subjectRef: common.Reference{
+				Path:   "localhost:5000/net-monitor",
+				Tag:    "v1",
+				Digest: testSubjectDigest,
+			},
+			store: &orasStore{},
+			repository: mocks.TestRepository{
+				ResolveMap: map[string]oci.Descriptor{
+					fmt.Sprintf("localhost:5000/net-monitor-not-found:%s", testCosignSubjectTag): {
+						Digest: testCosignImageDigest,
+					},
+				},
+			},
+			output: nil,
+			err:    nil,
+		},
+		{
+			name: "success: one cosign reference",
+			subjectRef: common.Reference{
+				Path:   "localhost:5000/net-monitor",
+				Tag:    "v1",
+				Digest: testSubjectDigest,
+			},
+			store: &orasStore{},
+			repository: mocks.TestRepository{
+				ResolveMap: map[string]oci.Descriptor{
+					fmt.Sprintf("localhost:5000/net-monitor:%s", testCosignSubjectTag): {
+						Digest: testCosignImageDigest,
+					},
+				},
+			},
+			output: &[]ocispecs.ReferenceDescriptor{
+				{
+					Descriptor: oci.Descriptor{
+						Digest: testCosignImageDigest,
+					},
+					ArtifactType: CosignArtifactType,
+				},
+			},
+			err: nil,
+		},
+	}
+	for _, testcase := range testcases {
+		refs, err := getCosignReferences(ctx, testcase.subjectRef, testcase.store, testcase.repository)
```

Comment:
not a blocker for this pr. We can have `t.Run()` with test name so that it will print out the name for each test. Example: https://github.com/notaryproject/notation-go/blob/main/registry/repository_test.go#L252

Reply:
switched to `t.Run()` thanks!